### ISSUE TITLE
[odt reader] bookmarks, references lead to duplicate entries

### ISSUE THREAD

**Issue by hubertp-lshift** [jgm/pandoc#3143]
```
<text:p text:style-name="P2">
  <text:span text:style-name="T1">
    Some text
    <text:line-break/>
    Another one with a link
  </text:span>
  <text:bookmark text:name="Simple bookmark"/>
</text:p>
```
will be parsed as (in native)
```
Para [Str "Some",Space,Str "text",LineBreak,Str "Another",Space,Str "one",Space,Str "with",Space,Str "a",Space,Str "link",Span ("anchor",[],[]) [],Str "Some",Space,Str "text",LineBreak,Str "Another",Space,Str "one",Space,Str "with",Space,Str "a",Space,Str "link"]
```

The problem is caused by the wrong definition of `maybeAddAnchorFrom` in
https://github.com/jgm/pandoc/blob/master/src/Text/Pandoc/Readers/Odt/ContentReader.hs#L717

**Comment by jgm** [comment]
@MarLinn can you take a look?

+++ hubertp-lshift [Oct 04 16 02:36 ]:

**Comment by hubertp-lshift** [comment]
For me
```@@ -714,7 +714,7 @@ maybeAddAnchorFrom anchorReader =
   proc (inlines, fAnchorElem) -> do
   case fAnchorElem of
     Right anchorElem ->
-      arr (anchorElem <>) -<< inlines
+      returnA -< anchorElem
     Left _ -> returnA -< inlines
   where
     toAnchorElem :: Anchor -> Inlines
``` did the trick but then my haskell foo is a bit rusty.

**Comment by jgm** [comment]
Closed by #3166
